src/Components/Web.JS/src/Platform/BootConfig.ts

```diff
@@ -14,9 +14,15 @@ export class BootConfigResult {
       loadBootResource('manifest', 'blazor.boot.json', '_framework/blazor.boot.json', '') :
       defaultLoadBlazorBootJson('_framework/blazor.boot.json');
 
-    const bootConfigResponse = loaderResponse instanceof Promise ?
-      await loaderResponse :
-      await defaultLoadBlazorBootJson(loaderResponse ?? '_framework/blazor.boot.json');
+    let bootConfigResponse: Response;
+
+    if (!loaderResponse) {
+      bootConfigResponse = await defaultLoadBlazorBootJson('_framework/blazor.boot.json');
+    } else if (typeof loaderResponse === 'string') {
+      bootConfigResponse = await defaultLoadBlazorBootJson(loaderResponse);
+    } else {
```

Comment:
Is there any further type enforcement we can do here, and then have an `else throw` case at the end?

Reply:
Oh got it, thanks for the clarification. This looks great in that case.